platform/initialize.go

```diff
@@ -0,0 +1,160 @@
+ * limitations under the License.
+ */
+
+package platform
+
+import (
+	"context"
+	"github.com/kiegroup/container-builder/api"
+	"github.com/kiegroup/container-builder/client"
+	v08 "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+	"github.com/kiegroup/kogito-serverless-operator/builder"
+	"github.com/kiegroup/kogito-serverless-operator/constants"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NewInitializeAction returns a action that initializes the platform configuration when not provided by the user.
+func NewInitializeAction() Action {
+	return &initializeAction{}
+}
+
+type initializeAction struct {
+	baseAction
+}
+
+func (action *initializeAction) Name() string {
+	return "initialize"
+}
+
+func (action *initializeAction) CanHandle(platform *v08.KogitoServerlessPlatform) bool {
+	return platform.Status.Phase == "" || platform.Status.Phase == v08.PlatformPhaseDuplicate
+}
+
+func (action *initializeAction) Handle(ctx context.Context, platform *v08.KogitoServerlessPlatform) (*v08.KogitoServerlessPlatform, error) {
+	duplicate, err := action.isPrimaryDuplicate(ctx, platform)
+	if err != nil {
+		return nil, err
+	}
+	if duplicate {
+		// another platform already present in the namespace
+		if platform.Status.Phase != v08.PlatformPhaseDuplicate {
+			platform := platform.DeepCopy()
+			platform.Status.Phase = v08.PlatformPhaseDuplicate
+
+			return platform, nil
+		}
+
+		return nil, nil
+	}
+
+	if err = ConfigureDefaults(ctx, action.client, platform, true); err != nil {
+		return nil, err
+	}
+	// nolint: staticcheck
+	if platform.Status.BuildPlatform.PublishStrategy == api.PlatformBuildPublishStrategyKaniko {
+		//If KanikoCache is enabled
+		if platform.Status.BuildPlatform.IsOptionEnabled(builder.KanikoBuildCacheEnabled) {
+			// Create the persistent volume claim used by the Kaniko cache
+			action.Logger.Info("Create persistent volume claim")
+			err := createPersistentVolumeClaim(ctx, action.client, platform)
+			if err != nil {
+				return nil, err
+			}
+			// Create the Kaniko warmer pod that caches the base image into the Kogito Serverless builder volume
+			action.Logger.Info("Create Kaniko cache warmer pod")
+			err = createKanikoCacheWarmerPod(ctx, action.client, platform)
+			if err != nil {
+				return nil, err
+			}
+			platform.Status.Phase = v08.PlatformPhaseWarming
+		} else {
+			// Skip the warmer pod creation
+			platform.Status.Phase = v08.PlatformPhaseCreating
+		}
+	} else {
+		platform.Status.Phase = v08.PlatformPhaseCreating
+	}
+	platform.Status.Version = constants.VERSION
+
+	return platform, nil
+}
+
+func createPersistentVolumeClaim(ctx context.Context, client client.Client, platform *v08.KogitoServerlessPlatform) error {
+	volumeSize, err := resource.ParseQuantity(constants.DEFAULT_KAKIKO_PVC_SIZE)
```

Comment:
Is this property exposed in the crd?

Reply:
Not at the moment

Reply:
ok, we can file a jira for future improvement.